docs/source/design/testing.rst

```diff
@@ -0,0 +1,99 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_testing:
+
+**********************************
+Continuous integration and testing
+**********************************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+DataLad is tested using a pytest-based testsuite that is run locally and via continuous integrations setups.
+Code development should ensure that old and new functionality is appropriately tested.
+The project aims for good unittest coverage (at least 80%).
+
+Running tests
+=============
+
+``datalad/tests`` contains tests for the core portion of the project.
+More tests are provided under corresponding submodules in ``tests/`` subdirectories to simplify re-running the tests concerning that portion of the codebase.
+The test suite is run using
+
+.. code-block:: bash
+
+   pytest datalad
```

Comment:
I recommend

```suggestion
   python -m pytest datalad
```

to ensure that the environment/python used for execution of datalad is used for testing.  Since otherwise, e.g. if no `pytest` was installed in the virtualenv, system-wide pytest could be used which would use system wide python:

```
❯ head -n1 /usr/bin/pytest
#!/usr/bin/python3
```

and test IIRC systemwide `datalad` and that  can cause all kinds of confusion.

But do not accept above -- below a more complete recipe